### PR TITLE
[SHELL32] Show file / drive properties at the position where the mouse was.

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-18386](https://jira.reactos.org/browse/CORE-18386)

### CODE REVIEW

File: dll/win32/shell32/CDefView.cpp [hunk 1]
```diff
@@ -1950,7 +1954,7 @@ LRESULT CDefView::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHand
             {
                 HMENU Dummy = NULL;
                 MenuCleanup _(m_pFileMenu, Dummy);
-                InvokeContextMenuCommand(m_pFileMenu, dwCmdID);
+                InvokeContextMenuCommand(m_pFileMenu, dwCmdID, NULL);
```
Comment (HBelusca) [inline dll/win32/shell32/CDefView.cpp:1957]: Does this show a menu or remove it? (and does a similar "FIXME" to the one above apply there too?)

File: dll/win32/shell32/dialogs/drive.cpp [hunk 2]
```diff
@@ -166,69 +166,69 @@ typedef struct _DRIVE_PROP_PAGE
...
+    psh.pszCaption = pwszDrive;
+    psh.hwndParent = stub;
+    psh.nStartPage = 0;
+    psh.phpage = hpsp;
 
+    HRESULT hr = CComObject<CDrvDefExt>::CreateInstance(&pDrvDefExt);
     if (SUCCEEDED(hr))
     {
-        hr = CComObject<CDrvDefExt>::CreateInstance(&pDrvDefExt);
+        pDrvDefExt->AddRef(); // CreateInstance returns object with 0 ref count
```
Comment (HBelusca) [inline dll/win32/shell32/dialogs/drive.cpp:207]: General question: Is this behaviour you mention in comment expected? (it's not a bug?)

File: dll/win32/shell32/dialogs/drive.cpp [hunk 3]
```diff
@@ -166,69 +166,69 @@ typedef struct _DRIVE_PROP_PAGE
...
-        }
-
-        hpsx = SHCreatePropSheetExtArrayEx(HKEY_CLASSES_ROOT, L"Drive", MAX_PROPERTY_SHEET_PAGE, pDataObj);
-        if (hpsx)
-            SHAddFromPropSheetExtArray(hpsx, (LPFNADDPROPSHEETPAGE)AddPropSheetPageCallback, (LPARAM)&psh);
+            hr = pDrvDefExt->AddPages(AddPropSheetPageCallback, (LPARAM)&psh);
+            if (FAILED(hr))
+                ERR("AddPages failed\n");
+        } else
+            ERR("Initialize failed\n");
```
Comment (HBelusca) [inline dll/win32/shell32/dialogs/drive.cpp:215]:
```suggestion
        }
        else
        {
            ERR("Initialize failed\n");
        }
```

File: dll/win32/shell32/dialogs/drive.cpp (hunk 2 above)
Comment (learn-more) [reply dll/win32/shell32/dialogs/drive.cpp:207]: It is expected for `CComObject`, but unexpected in normal COM behavior, hence the comment.

File: dll/win32/shell32/dialogs/drive.cpp (hunk 3 above)
Comment (learn-more) [reply dll/win32/shell32/dialogs/drive.cpp:215]: Only moved this, but sure.

File: dll/win32/shell32/CDefView.cpp (hunk 1 above)
Comment (learn-more) [reply dll/win32/shell32/CDefView.cpp:1957]: As the comment indicates, this is the 'File' menu.

**Review by JoachimHenze — APPROVED** [review]

**Comment by binarymaster** [comment]
Houston we have problem: https://chat.reactos.org/reactos/pl/cs1uyrxmzpd7mf4oo57q13wq6a

Started to happen after this PR.

**Pull request merged**